crates/uv/src/commands/python/pin.rs

```diff
@@ -124,33 +130,48 @@ pub(crate) async fn pin(
         request
     };
 
-    let existing = version_file.ok().flatten();
     // TODO(zanieb): Allow updating the discovered version file with an `--update` flag.
-    let new = PythonVersionFile::new(project_dir.join(PYTHON_VERSION_FILENAME))
-        .with_versions(vec![request]);
-
-    new.write().await?;
+    if let Some(existing) = version_file {
+        if existing.contains(&request) {
+            writeln!(
+                printer.stdout(),
+                "`{}` is already pinned at `{}`",
+                existing.path().user_display().cyan(),
+                request.to_canonical_string().green(),
+            )?;
+            return Ok(ExitStatus::Success);
+        }
 
-    if let Some(existing) = existing
-        .as_ref()
-        .and_then(PythonVersionFile::version)
-        .filter(|version| *version != new.version().unwrap())
-    {
+        let new = existing.clone().with_versions(vec![request]);
```

Comment:
Behavior change: `uv python pin` now updates the discovered version file, rather than always writing to `.python-version`.